File: platform-console/apps/raap-admin/src/api/core/expert-debug.ts
/**
 * Expert 调试 API
 */
import { requestClient } from '#/api/request';

// ==================== 类型定义 ====================

export namespace ExpertDebugApi {
  /** Token 使用情况 */
  export interface TokenUsage {
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
  }

  /** Plugin 配置快照项 */
  export interface PluginConfigSnapshotItem {
    plugin_code: string;
    variable_mapping: Record<string, string>;
  }

  /** 调试请求 */
  export interface DebugRequest {
    expert_config_code: string;
    content: string;
    plugin_config_snapshot?: PluginConfigSnapshotItem[];
    model_code?: string;
    model_config_override?: Record<string, any>;
    prompt_override?: string;
  }

  /** 调试响应 */
  export interface DebugResponse {
    id?: number;
    success: boolean;
    expert_config_code: string;
    expert_config_name?: string;
    model_code?: string;
    model_config_used?: Record<string, any>;
    prompt_template?: string;
    plugin_config_snapshot?: PluginConfigSnapshotItem[];
    rendered_prompt?: string;
    prompt_override?: string;
    input_content: string;
    output_content?: string;
    expert_total_output?: Record<string, any>; // Expert 返回的完整结果
    execution_time_ms: number;
    token_usage?: TokenUsage;
    error_message?: string;
    trace_id?: string;
    create_time?: string;
  }

  /** 预览 Prompt 请求 */
  export interface PreviewPromptRequest {
    expert_config_code: string;
    plugin_config_snapshot?: PluginConfigSnapshotItem[];
  }

  /** 插件渲染段 */
  export interface PluginSegment {
    plugin_code: string;
    plugin_name?: string;
    content: string;
  }

  /** 预览 Prompt 响应 */
  export interface PreviewPromptResponse {
    expert_config_code: string;
    prompt_template?: string;
    plugin_config?: PluginConfigSnapshotItem[];
    plugin_config_snapshot?: PluginConfigSnapshotItem[];
    rendered_prompt: string;
    plugin_segments: PluginSegment[];
    variables_used: Record<string, string>;
  }

  /** Plugin 变量选项 */
  export interface PluginVariableOption {
    context_name: string;
    context_preview?: string;
  }

  /** 用户画像配置 */
  export interface UserProfileConfig {
    source: 'user_profile';
    label?: string;
    strategy?: 'all' | 'random' | 'weighted';
  }

  /** 策略节点信息（含语料预览） */
  export interface StrategyNodeInfo {
    node_id: string;
    node_name: string;
    corpus_count: number;
    corpus_preview?: string;
  }

  /** 策略绑定信息 */
  export interface StrategyInfo {
    strategy_id: number;
    strategy_name: string;
    label: string;
    node_count: number;
  }

  /** Plugin 变量 */
  export interface PluginVariable {
    variable_name: string;
    options: PluginVariableOption[];
    selected?: string;
    // 模式字段
    source?: 'plugin_context' | 'strategy';
    // 策略绑定模式字段
    strategy_info?: StrategyInfo;
    strategy_nodes?: StrategyNodeInfo[];
  }

  /** Plugin 变量响应 */
  export interface PluginVariablesResponse {
    plugin_code: string;
    plugin_name?: string;
    variables: PluginVariable[];
    strategy_info?: {
      label: string;
      node_count: number;
      strategy_id: number;
      strategy_name: string;
    };
  }

  /** Expert Plugin 变量响应 */
  export interface ExpertPluginVariablesResponse {
    expert_config_code: string;
    plugins: PluginVariablesResponse[];
  }

  /** 调试历史列表响应 */
  export interface DebugHistoryListResponse {
    items: DebugResponse[];
    total: number;
    page: number;
    page_size: number;
  }

  /** 历史列表查询参数 */
  export interface HistoryListParams {
    expert_config_code?: string;
    success?: boolean;
    is_starred?: boolean;
    page?: number;
    page_size?: number;
  }

  /** 批量随机调试请求 */
  export interface BatchDebugRequest {
    expert_config_code: string;
    content?: string;
    count?: number;
    include_current?: boolean;
    current_plugin_config_snapshot?: PluginConfigSnapshotItem[];
    model_code?: string;
    model_config_override?: Record<string, any>;
    prompt_override?: string;
  }

  /** 批量调试单条结果 */
  export interface BatchDebugResultItem {
    index: number;
    success: boolean;
    plugin_config_snapshot?: PluginConfigSnapshotItem[];
    variable_summary: string;
    title?: string;
    output_preview: string;
    output_content?: string;
    execution_time_ms: number;
    error_message?: string;
    history_id?: number;
  }

  /** 批量随机调试响应 */
  export interface BatchDebugResponse {
    expert_config_code: string;
    expert_config_name?: string;
    total: number;
    success_count: number;
    failed_count: number;
    total_time_ms: number;
    results: BatchDebugResultItem[];
  }

  /** 批量调试任务创建响应（异步模式） */
  export interface BatchDebugTaskResponse {
    task_id: string;
    status: string;
    expert_config_code: string;
    total: number;
    message: string;
  }

  /** 批量调试任务状态响应 */
  export interface BatchDebugTaskStatusResponse {
    task_id: string;
    status: string;
    expert_config_code: string;
    expert_config_name?: string;
    total: number;
    completed: number;
    success_count: number;
    failed_count: number;
    results: BatchDebugResultItem[];
    error_message?: string;
    start_time?: string;
    end_time?: string;
    create_time?: string;
  }

  /** SSE 进度事件数据 */
  export interface BatchDebugProgressEvent {
    task_id: string;
    status: string;
    total: number;
    completed: number;
    success_count: number;
    failed_count: number;
    results: BatchDebugResultItem[];
  }

  /** 批量调试任务列表项 */
  export interface BatchDebugTaskListItem {
    id: number;
    task_id: string;
    expert_config_code: string;
    expert_config_name?: string;
    status: string;
    total: number;
    completed: number;
    success_count: number;
    failed_count: number;
    start_time?: string;
    end_time?: string;
    create_time: string;
  }

  /** 批量调试任务历史列表响应 */
  export interface BatchDebugTaskListResponse {
    items: BatchDebugTaskListItem[];
    total: number;
    page: number;
    page_size: number;
  }
}

// ==================== API 函数 ====================

/**
 * 执行 Expert 调试
 */
export async function debugExpertApi(data: ExpertDebugApi.DebugRequest) {
  return requestClient.post<ExpertDebugApi.DebugResponse>(
    '/v1/expert-configs/debug',
    data,
    {
      timeout: 180_000, // 3 分钟超时
    },
  );
}

/**
 * 预览 Prompt 渲染结果
 */
export async function previewPromptApi(
  data: ExpertDebugApi.PreviewPromptRequest,
) {
  return requestClient.post<ExpertDebugApi.PreviewPromptResponse>(
    '/v1/expert-configs/preview-prompt',
    data,
  );
}

/**
 * 获取 Expert 关联的 Plugin 变量选项
 */
export async function getPluginVariablesApi(expertConfigCode: string) {
  return requestClient.get<ExpertDebugApi.ExpertPluginVariablesResponse>(
    `/v1/expert-configs/${expertConfigCode}/plugin-variables`,
    {
      // 添加时间戳参数避免浏览器缓存
      params: {
        _t: Date.now(),
      },
    },
  );
}

/**
 * 获取调试历史列表
 */
export async function getDebugHistoryApi(
  params?: ExpertDebugApi.HistoryListParams,
) {
  return requestClient.get<ExpertDebugApi.DebugHistoryListResponse>(
    '/v1/expert-configs/debug-history',
    { params },
  );
}

/**
 * 获取单条调试历史详情
 */
export async function getDebugHistoryDetailApi(historyId: number) {
  return requestClient.get<ExpertDebugApi.DebugResponse>(
    `/v1/expert-configs/debug-history/${historyId}`,
  );
}

/**
 * 收藏/取消收藏调试历史
 */
export async function starDebugHistoryApi(
  historyId: number,
  isStarred: boolean,
) {
  return requestClient.put(
    `/v1/expert-configs/debug-history/${historyId}/star`,
    {
      is_starred: isStarred,
    },
  );
}

/**
 * 删除调试历史
 */
export async function deleteDebugHistoryApi(historyId: number) {
  return requestClient.delete(`/v1/expert-configs/debug-history/${historyId}`);
}

/**
 * 批量随机调试（异步模式）
 * 立即返回 task_id，通过 SSE 接收实时进度
 */
export async function batchDebugExpertApi(
  data: ExpertDebugApi.BatchDebugRequest,
) {
  return requestClient.post<ExpertDebugApi.BatchDebugTaskResponse>(
    '/v1/expert-configs/debug-batch',
    data,
  );
}

/**
 * 查询批量调试任务状态
 */
export async function getBatchDebugTaskStatusApi(taskId: string) {
  return requestClient.get<ExpertDebugApi.BatchDebugTaskStatusResponse>(
    `/v1/expert-configs/debug-batch/${taskId}`,
  );
}

/**
 * 获取批量调试任务历史列表
 */
export async function listBatchDebugTasksApi(params?: {
  expert_config_code?: string;
  page?: number;
  page_size?: number;
  status?: string;
}) {
  return requestClient.get<ExpertDebugApi.BatchDebugTaskListResponse>(
    '/v1/expert-configs/debug-batch',
    { params },
  );
}

// ==================== 批量评分（test_case / expert_eval_*） ====================

export namespace ExpertEvalApi {
  /** 测试集选项（用于下拉选择） */
  export interface TestSetOption {
    id: number;
    code: string;
    name: string;
    type: 'image' | 'text';
    case_count?: number;
  }

  export interface TestCaseItem {
    id: number;
    test_set_code: string;
    title?: null | string;
    content?: null | string;
    image_url?: null | string;
    enabled: number;
    create_time?: string;
    update_time?: string;
  }

  export interface TestCaseListResponse {
    items: TestCaseItem[];
    total: number;
    page: number;
    page_size: number;
  }

  export interface EvalRunItem {
    id: number;
    run_code: string;
    expert_config_code: string;
    test_set_code?: null | string;
    status: 'cancelled' | 'failed' | 'running' | 'success';
    total_count: number;
    success_count: number;
    failed_count: number;
    start_time?: string;
    end_time?: null | string;
    created_by?: null | string;
  }

  export interface EvalRunListResponse {
    items: EvalRunItem[];
    total: number;
    page: number;
    page_size: number;
  }

  export interface CreateEvalRunRequest {
    expert_config_code: string;
    test_set_code?: string;
    test_case_ids?: number[];
    max_count?: number;
    start_no?: number;
    end_no?: number;
    // 前端可选：并发/超时等，后端可忽略
    article_concurrency?: number;
  }

  export interface CreateEvalRunResponse {
    run_id: number;
    run_code: string;
  }

  export interface EvalResultItem {
    id: number;
    run_id: number;
    test_case_id: number;
    score?: null | number;
    reason?: null | string;
    highlights?: null | string;
    problem_tags?: null | string[];
    problem_snippets?: null | string[];
    problem_context_list?: null | string[];
    success: boolean;
    error_message?: null | string;
    latency_ms?: null | number;
    model_code?: null | string;
    trace_id?: null | string;
    create_time?: string;
  }

  export interface EvalResultListResponse {
    items: EvalResultItem[];
    total: number;
    page: number;
    page_size: number;
  }

  export interface EvalResultDetailTestCase {
    id: number;
    test_set_code: string;
    title?: null | string;
    content?: null | string;
    image_url?: null | string;
  }

  export interface EvalResultDetail {
    id: number;
    run_id: number;
    test_case_id: number;
    score?: null | number;
    reason?: null | string;
    highlights?: null | string;
    problem_tags?: null | string[];
    problem_snippets?: null | string[];
    problem_context_list?: null | string[];
    success: boolean;
    error_message?: null | string;
    latency_ms?: null | number;
    model_code?: null | string;
    provider_code?: null | string;
    token_usage?: null | Record<string, any>;
    trace_id?: null | string;
    rendered_prompt?: null | string;
    raw_output?: null | Record<string, any>;
    create_time?: string;
    test_case?: EvalResultDetailTestCase | null;
  }
}

/**
 * 获取测试用例列表（test_case）
 */
export async function listTestCasesApi(params?: {
  enabled?: boolean;
  keyword?: string;
  page?: number;
  page_size?: number;
  test_set_code?: string;
}) {
  return requestClient.get<ExpertEvalApi.TestCaseListResponse>(
    '/v1/test-cases',
    {
      params,
    },
  );
}

/**
 * 获取测试集选项列表（用于批量评分下拉）
 */
export async function listTestSetOptionsApi() {
  return requestClient.get<ExpertEvalApi.TestSetOption[]>(
    '/v1/test-sets/options',
  );
}

/**
 * 创建并触发一次评测 run（后端应从 test_case 取数并写入 expert_eval_result）
 */
export async function createEvalRunApi(
  data: ExpertEvalApi.CreateEvalRunRequest,
) {
  return requestClient.post<ExpertEvalApi.CreateEvalRunResponse>(
    '/v1/expert-evals/runs',
    data,
    {
      timeout: 600_000, // 10 分钟
    },
  );
}

/**
 * 查询评测 run 列表
 */
export async function listEvalRunsApi(params?: {
  expert_config_code?: string;
  page?: number;
  page_size?: number;
  status?: string;
  test_set_code?: string;
}) {
  return requestClient.get<ExpertEvalApi.EvalRunListResponse>(
    '/v1/expert-evals/runs',
    { params },
  );
}

/**
 * 查询评测结果列表
 */
export async function listEvalResultsApi(params: {
  page?: number;
  page_size?: number;
  run_id: number;
  success?: boolean;
}) {
  return requestClient.get<ExpertEvalApi.EvalResultListResponse>(
    '/v1/expert-evals/results',
    { params },
  );
}

/**
 * 查询单条评测结果详情（用于抽屉/弹窗）
 */
export async function getEvalResultDetailApi(resultId: number) {
  return requestClient.get<ExpertEvalApi.EvalResultDetail>(
    `/v1/expert-evals/results/${resultId}`,
  );
}

File: platform-server/app/schemas/expert_debug.py
"""
Expert 调试相关的 Schema
"""
from datetime import datetime
from typing import Optional, Dict, Any, List

from pydantic import Field

from app.schemas.base import BaseSchema


class ExpertDebugRequest(BaseSchema):
    """Expert 调试请求"""
    expert_config_code: str = Field(..., description="要调试的 expert_config_code")
    content: str = Field(..., description="测试内容")
    plugin_config_snapshot: Optional[List[Dict[str, Any]]] = Field(
        default=None, 
        description="可选：覆盖 plugin_config_snapshot，格式: [{plugin_code, variable_mapping}]"
    )
    model_code: Optional[str] = Field(
        default=None, 
        description="可选：覆盖模型编码"
    )
    model_cfg_override: Optional[Dict[str, Any]] = Field(
        default=None,
        alias="model_config_override",
        description="可选：覆盖模型配置 (temperature, max_tokens 等)"
    )
    prompt_override: Optional[str] = Field(
        default=None, 
        description="可选：完全覆盖渲染后的 Prompt"
    )


class TokenUsage(BaseSchema):
    """Token 使用情况"""
    prompt_tokens: int = Field(default=0, description="Prompt tokens")
    completion_tokens: int = Field(default=0, description="Completion tokens")
    total_tokens: int = Field(default=0, description="Total tokens")


class ExpertDebugResponse(BaseSchema):
    """Expert 调试响应"""
    id: Optional[int] = Field(default=None, description="调试记录 ID")
    success: bool = Field(..., description="执行是否成功")
    expert_config_code: str = Field(..., description="Expert 配置编码")
    expert_config_name: Optional[str] = Field(default=None, description="Expert 配置名称")
    model_code: Optional[str] = Field(default=None, description="使用的模型")
    model_config_used: Optional[Dict[str, Any]] = Field(default=None, description="实际使用的模型配置")
    prompt_template: Optional[str] = Field(default=None, description="原始 Prompt 模板")
    plugin_config_snapshot: Optional[List[Dict[str, Any]]] = Field(default=None, description="使用的变量快照，格式: [{plugin_code, variable_mapping}]")
    rendered_prompt: Optional[str] = Field(default=None, description="渲染后的 Prompt")
    prompt_override: Optional[str] = Field(default=None, description="用户覆盖的 Prompt")
    input_content: str = Field(..., description="输入内容")
    output_content: Optional[str] = Field(default=None, description="AI 输出内容（主要内容）")
    expert_total_output: Optional[Dict[str, Any]] = Field(default=None, description="Expert 返回的完整结果")
    execution_time_ms: int = Field(default=0, description="执行时间(毫秒)")
    token_usage: Optional[TokenUsage] = Field(default=None, description="Token 使用情况")
    error_message: Optional[str] = Field(default=None, description="错误信息")
    trace_id: Optional[str] = Field(default=None, description="追踪 ID")
    create_time: Optional[datetime] = Field(default=None, description="创建时间")
    is_starred: bool = Field(default=False, description="是否收藏")


class PreviewPromptRequest(BaseSchema):
    """预览 Prompt 渲染请求"""
    expert_config_code: str = Field(..., description="Expert 配置编码")
    plugin_config_snapshot: Optional[List[Dict[str, Any]]] = Field(
        default=None, 
        description="可选：指定变量快照，格式: [{plugin_code, variable_mapping}]，不传则随机选择"
    )


class PluginSegment(BaseSchema):
    """插件渲染段"""
    plugin_code: str = Field(..., description="Plugin 编码")
    plugin_name: Optional[str] = Field(default=None, description="Plugin 名称")
    content: str = Field(..., description="该插件渲染的内容")


class PreviewPromptResponse(BaseSchema):
    """预览 Prompt 渲染响应"""
    expert_config_code: str = Field(..., description="Expert 配置编码")
    prompt_template: Optional[str] = Field(default=None, description="原始 Prompt 模板")
    plugin_config: Optional[List[Dict[str, Any]]] = Field(default=None, description="原始 plugin_config 数组")
    plugin_config_snapshot: Optional[List[Dict[str, Any]]] = Field(default=None, description="使用的变量快照，格式: [{plugin_code, variable_mapping}]")
    rendered_prompt: str = Field(..., description="渲染后的 Prompt")
    plugin_segments: List[PluginSegment] = Field(default_factory=list, description="插件分段信息，按顺序显示每个插件渲染的内容")
    variables_used: Dict[str, str] = Field(default_factory=dict, description="使用的变量映射")


class PluginVariableOption(BaseSchema):
    """Plugin 变量选项"""
    context_name: str = Field(..., description="上下文名称/节点名称")
    context_preview: Optional[str] = Field(default=None, description="内容预览")
    node_id: Optional[str] = Field(default=None, description="节点 ID（关键词树模式）")


class KeywordTreeNode(BaseSchema):
    """关键词树节点"""
    node_id: str = Field(..., description="节点 ID")
    node_name: str = Field(..., description="节点名称")
    label: Optional[str] = Field(default=None, description="节点 label")
    corpus_count: int = Field(default=0, description="语料数量")


class StrategyNodeInfo(BaseSchema):
    """策略节点信息（含语料预览）"""
    node_id: str = Field(..., description="节点 ID")
    node_name: str = Field(..., description="节点名称")
    corpus_count: int = Field(default=0, description="语料数量")
    corpus_preview: Optional[str] = Field(default=None, description="语料内容预览")
    select_mode: Optional[str] = Field(default=None, description="节点选择模式：single-分开使用 / multiple-合在一起使用")


class StrategyInfo(BaseSchema):
    """策略绑定信息"""
    strategy_id: int = Field(..., description="策略 ID")
    strategy_name: str = Field(..., description="策略名称")
    label: str = Field(..., description="映射的维度标签")
    node_count: int = Field(default=0, description="可选节点数量")


class PluginVariable(BaseSchema):
    """Plugin 变量"""
    variable_name: str = Field(..., description="变量名")
    source: str = Field(default="plugin_context", description="变量来源: plugin_context, keyword_tree, strategy")
    # 旧模式 (plugin_context) 相关字段
    options: List[PluginVariableOption] = Field(default_factory=list, description="可选值列表（plugin_context 模式）")
    selected: Optional[str] = Field(default=None, description="当前选中的值（plugin_context 模式）")
    # keyword_tree 模式相关字段
    keyword_tree_config: Optional[Dict[str, Any]] = Field(default=None, description="关键词树配置（keyword_tree 模式）")
    keyword_tree_nodes: List[KeywordTreeNode] = Field(default_factory=list, description="已选节点列表（keyword_tree 模式）")
    strategy: Optional[str] = Field(default=None, description="选择策略: random, weighted, all")
    # 策略绑定模式 (strategy) 相关字段
    strategy_info: Optional[StrategyInfo] = Field(default=None, description="策略绑定信息")
    strategy_nodes: List[StrategyNodeInfo] = Field(default_factory=list, description="可选节点列表（策略模式）")


class PluginVariablesResponse(BaseSchema):
    """Plugin 变量列表响应"""
    plugin_code: str = Field(..., description="Plugin 编码")
    plugin_name: Optional[str] = Field(default=None, description="Plugin 名称")
    variables: List[PluginVariable] = Field(default_factory=list, description="变量列表")
    strategy_info: Optional[StrategyInfo] = Field(default=None, description="策略绑定信息（插件级别）")


class ExpertPluginVariablesResponse(BaseSchema):
    """Expert 关联的所有 Plugin 变量"""
    expert_config_code: str = Field(..., description="Expert 配置编码")
    plugins: List[PluginVariablesResponse] = Field(default_factory=list, description="Plugin 列表")


class DebugHistoryListRequest(BaseSchema):
    """调试历史列表请求"""
    expert_config_code: Optional[str] = Field(default=None, description="筛选特定 Expert")
    success: Optional[bool] = Field(default=None, description="筛选成功/失败")
    is_starred: Optional[bool] = Field(default=None, description="筛选收藏")
    page: int = Field(default=1, ge=1, description="页码")
    page_size: int = Field(default=20, ge=1, le=100, description="每页数量")


class DebugHistoryListResponse(BaseSchema):
    """调试历史列表响应"""
    items: List[ExpertDebugResponse] = Field(default_factory=list, description="历史记录列表")
    total: int = Field(default=0, description="总数")
    page: int = Field(default=1, description="当前页")
    page_size: int = Field(default=20, description="每页数量")


class StarHistoryRequest(BaseSchema):
    """收藏/取消收藏请求"""
    is_starred: bool = Field(..., description="是否收藏")


class BatchDebugRequest(BaseSchema):
    """批量随机调试请求"""
    expert_config_code: str = Field(..., description="要调试的 expert_config_code")
    content: str = Field(default="", description="测试内容（CRITIC 类型用）")
    count: int = Field(default=5, ge=1, le=20, description="执行次数（1-20）")
    include_current: bool = Field(default=True, description="是否将当前变量作为首次测试")
    current_plugin_config_snapshot: Optional[List[Dict[str, Any]]] = Field(
        default=None,
        description="当前变量快照（include_current=true 时使用）"
    )
    model_code: Optional[str] = Field(default=None, description="可选：覆盖模型编码")
    model_cfg_override: Optional[Dict[str, Any]] = Field(
        default=None,
        alias="model_config_override",
        description="可选：覆盖模型配置"
    )
    prompt_override: Optional[str] = Field(default=None, description="可选：覆盖 Prompt")


class BatchDebugResultItem(BaseSchema):
    """批量调试单条结果"""
    index: int = Field(..., description="执行序号（从 1 开始）")
    success: bool = Field(..., description="执行是否成功")
    plugin_config_snapshot: Optional[List[Dict[str, Any]]] = Field(
        default=None,
        description="使用的变量快照"
    )
    variable_summary: str = Field(default="", description="变量组合摘要")
    title: str = Field(default="", description="生成内容的标题")
    output_preview: str = Field(default="", description="输出预览（前 200 字）")
    output_content: Optional[str] = Field(default=None, description="完整输出")
    execution_time_ms: int = Field(default=0, description="执行时间(毫秒)")
    error_message: Optional[str] = Field(default=None, description="错误信息")
    history_id: Optional[int] = Field(default=None, description="历史记录 ID")


class BatchDebugResponse(BaseSchema):
    """批量随机调试响应"""
    expert_config_code: str = Field(..., description="Expert 配置编码")
    expert_config_name: Optional[str] = Field(default=None, description="Expert 配置名称")
    total: int = Field(default=0, description="执行总数")
    success_count: int = Field(default=0, description="成功数")
    failed_count: int = Field(default=0, description="失败数")
    total_time_ms: int = Field(default=0, description="总耗时(毫秒)")
    results: List[BatchDebugResultItem] = Field(default_factory=list, description="结果列表")


class BatchDebugTaskResponse(BaseSchema):
    """批量调试任务创建响应（异步模式）"""
    task_id: str = Field(..., description="任务 ID")
    status: str = Field(..., description="任务状态: pending/running/completed/failed")
    expert_config_code: str = Field(..., description="Expert 配置编码")
    total: int = Field(..., description="总任务数")
    message: str = Field(..., description="提示信息")


class BatchDebugTaskStatusResponse(BaseSchema):
    """批量调试任务状态响应"""
    task_id: str = Field(..., description="任务 ID")
    status: str = Field(..., description="任务状态: pending/running/completed/failed")
    expert_config_code: str = Field(..., description="Expert 配置编码")
    expert_config_name: Optional[str] = Field(default=None, description="Expert 配置名称")
    total: int = Field(default=0, description="总任务数")
    completed: int = Field(default=0, description="已完成数")
    success_count: int = Field(default=0, description="成功数")
    failed_count: int = Field(default=0, description="失败数")
    results: List[BatchDebugResultItem] = Field(default_factory=list, description="结果列表")
    error_message: Optional[str] = Field(default=None, description="错误信息")
    start_time: Optional[datetime] = Field(default=None, description="开始时间")
    end_time: Optional[datetime] = Field(default=None, description="结束时间")
    create_time: Optional[datetime] = Field(default=None, description="创建时间")

